src/cgi/CgiHandler.cpp

```diff
@@ -71,17 +75,109 @@ void	CgiHandler::handleRead( void )
 
 void	CgiHandler::handleWrite( void )
 {
-	const int	bytesLeft = _cgiInput.size() - _bytesWrote;
-	int			bytesToWrite = WRITE_SIZE;
-	int			currentWriteAmount;
+	size_t	bytesToWrite = WRITE_SIZE;
+
+	if (bytesToWrite > _cgiInput.size())
+		bytesToWrite = _cgiInput.size();
+
+	ssize_t bytesSent = write(_pipeWrite, _cgiInput.data(), bytesToWrite);
+	if (bytesSent < 0)
+		throw ( std::runtime_error("Failed to send response to client") );
 
-	if (bytesLeft < WRITE_SIZE)
-		bytesToWrite = bytesLeft;
+	_cgiInput.erase(0, bytesSent);
+
+	if (_cgiInput.size() == 0) {
+		// wrote everything to pipe
+		addKqueueEventFilter( _kqueueFd, _pipeRead, EVFILT_READ );
+		return ;
+	}
+	addKqueueEventFilter( _kqueueFd, _pipeWrite, EVFILT_WRITE );
+}
+
+/**
+ * @brief Starts a Python CGI script. 
+ * @note 2 pipes are created.
+ * In the child process we close the write end pipeToCgi[1] and the read end pipeFromCgi[0].
+ * In the parent process we close the read end pipeToCgi[0] and the write end pipeFromCgi[1].
+ * pipeToCgi is used to send data to the CGI script (parent to child). 
+ * pipeFromCgi is used to receive data from the CGI script (child to parent).
+ * Result:
+ *  ________                 _______
+ * |        |  pipeToCgi    |       |
+ * |        |  >--------->  |       |
+ * | Parent |               | Child |
+ * |        |  pipeFromCgi  |       |
+ * |________|  <---------<  |_______|            
+ */
+void	CgiHandler::startPythonCgi( void )
+{
+	int pipeToCgi[2];
+	int pipeFromCgi[2];
+	
+	// temp
+	char *args[] = { "python", "/Users/cpost/Desktop/webserv/src/test.py", NULL };
+	
+	// Init pipes. Throws runtime_error on failure.
+	if ( pipe( pipeToCgi ) == -1 || pipe( pipeFromCgi ) == -1 )
+		throw ( std::runtime_error("Failed to create pipe") );
+		
+	// Fork process. Throws runtime_error on failure.
+	this->_forkPid = fork();
+	if ( this->_forkPid == -1 )
+		throw ( std::runtime_error( "Failed to fork process" ) );
+	else if ( this->_forkPid == 0 ) // Child process
+	{
+		// Setup pipes in child process
+		this->childInitPipes( pipeToCgi, pipeFromCgi );
 
-	currentWriteAmount = write(_pipeWrite, &(_cgiInput[_bytesWrote]), bytesToWrite);
-	if (currentWriteAmount < 0)
-		throw ( std::runtime_error("Failed to write to the CGI") );
-	_bytesWrote += currentWriteAmount;
+		// Execute the CGI script
+    	char *env[] = { NULL };
+		execve( PYTHON_PATH, args, env );
+		std::cerr << "Error executing Python script" << std::endl;
+		return ;
```

Comment:
Since this is the child process we want to exit after an execve fail
```suggestion
	exit(1);
```